File: Scala/src/main/scala/PrimaryClasses/K08.scala
package PrimaryClasses
import IExecutable.IExecutable
import K08_1sub.GetMaxValue
import K08_1sub.GetMinValue
import K08_2sub.AttackGolem

import scala.collection.mutable
import scala.util.Random

class K08 extends IExecutable {
  override def Question1(): Unit = {
    var numbers = mutable.MutableList[Int]()
    try {
      for (i <- 1 to 3) {
        printf("%dつ目の値を入力してください。＞", i)
        numbers += scala.io.StdIn.readInt()
      }
      println("どちらを調べますか？")
      print("（０：最大値　１：最小値）＞")
      val choice = scala.io.StdIn.readInt()
      choice match {
        case 0 => printf("%dの中で最大値は%dです。", numbers.length, GetMaxValue(numbers.head, numbers(1), numbers(2)))
        case 1 => printf("%dの中で最小値は%dです。", numbers.length, GetMinValue(numbers.head, numbers(1), numbers(2)))
        case _ => ()
      }
    } catch {
      case ex: Exception => System.err.println(ex.getMessage)
    }
  }

  override def Question2(): Unit = {
    try {
      println("冒険が今始まる！")
      var playerHp = 200 + Random.nextInt(101)
      var continueGame = 1
      while (continueGame != 0) {
        if (playerHp == 0)
          return

        printf("あなたのHP：%d\n", playerHp)
        print("奥に進みますか？（１：奥に進む　０．帰る）＞")
        continueGame = scala.io.StdIn.readInt()
        if (continueGame == 1) {
          val golemLv = Random.nextInt(5)
          playerHp = AttackGolem(golemLv, playerHp)
        }
      }
      println("リレ〇ト！")
    } catch {
      case ex: Exception => System.err.println(ex.getMessage)
    }
  }

  override def Question3(): Unit = {

  }

  override def Question4(): Unit = {

  }
}

File: Java/src/com/ecc/gkgo1a/K08.java
package com.ecc.gkgo1a;

import java.util.Random;
import java.util.Scanner;
import java.util.Vector;

public class K08 extends IExecutable {

    @Override
    public void Question1() {
        Vector<Integer> numbers = new Vector<>();
        Scanner in = new Scanner(System.in);
        try {
            for (int i = 0; i < 3; i++) {
                System.out.print(String.format("%dつ目の値を入力してください。＞", i + 1));
                numbers.add(in.nextInt());
            }
            System.out.println("どちらを調べますか？");
            System.out.print("（０：最大値　１：最小値）＞");
            int choice = in.nextInt();
            switch (choice) {
                case 0:
                    System.out.println(String.format("%dの中で最大値は%dです。", numbers.size(),
                            K08_1Sub.GetMaxValue(numbers.get(0), numbers.get(1), numbers.get(2))));
                    break;
                case 1:
                    System.out.println(String.format("%dの中で最小値は%dです。", numbers.size(),
                            K08_1Sub.GetMinValue(numbers.get(0), numbers.get(1), numbers.get(2))));
                    break;
            }
        }
        catch (Exception ex) {
            System.err.println(ex.getMessage());
        }
    }

    @Override
    public void Question2() {
        try {
            System.out.println("冒険が今始まる！");
            Random random = new Random();
            int playerHp = 200 + random.nextInt(101);
            int continueGame = 1;
            Scanner in = new Scanner(System.in);
            do {
                if (playerHp == 0)
                    return;

                System.out.println("あなたのHP：" + playerHp);
                System.out.print("奥に進みますか？（１：奥に進む　０．帰る）＞");
                continueGame = in.nextInt();
                if (continueGame == 1) {
                    int golemLv = random.nextInt(5);
                    playerHp = K08_2Sub.AttackGolem(golemLv, playerHp);
                }
            } while (continueGame != 0);
            System.out.println("リレ〇ト！");
        }
        catch (Exception ex) {
            System.err.println(ex.getMessage());
        }
    }

    @Override
    public void Question3() {
    }

    @Override
    public void Question4() {
    }
}
